Add `if_only` option to rule conditional_returns_on_newline

Dschee opened the pull request:
Fixes #2307.

marcelofabri commented on Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewlineRule.swift:
```diff
@@ -30,19 +30,21 @@ public struct ConditionalReturnsOnNewlineRule: ConfigurationProviderRule, Rule,
 
     public func validate(file: File) -> [StyleViolation] {
-        let pattern = "(guard|if)[^\n]*return"
+        let pattern: String = configuration.ifOnly ? "(if)[^\n]*return" : "(guard|if)[^\n]*return"
```
we usually don't add explicit types

Dschee replied:
Removed.

marcelofabri commented on Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewlineRule.swift:
```diff
@@ -30,19 +30,21 @@ public struct ConditionalReturnsOnNewlineRule: ConfigurationProviderRule, Rule,
             }
 
-            return ["if", "guard"].contains(content(for: firstToken, file: file)) &&
+            let searchTokens: [String] = configuration.ifOnly ? ["if"] : ["if", "guard"]
```
we usually don't add explicit types

Dschee replied:
This was merely to speed up the compiler since it seems to be slow on type-checking ternary operators. But I can remove it, if you want.

Dschee replied:
Removed.

Dschee commented:
@marcelofabri I just removed the `AutomaticTestableRule` and introduced a manual test which covers the configuration as suggested. I hope it looks good now!
